### PR TITLE
Frontend Validations and Prototype Integration

### PR DESCRIPTION
**Pull request by DavidEBest**
* Improved frontend validation code.
* Refactored ui-router to remove unnecessary level of hierarchy
* Added a proposal report page, and refactored some of the proposal and report code.

### CODE REVIEW

File: sbirez/static/js/services/proposalsvc.js [hunk 1]
```diff
@@ -67,7 +79,17 @@ angular.module('sbirezApp').factory('ProposalService', function($http, $window,
 
   var saveProposalData = function() {
     var deferred = $q.defer();
-    $http.patch(PROPOSAL_URI + proposal.id + '/partial/', {'data':JSON.stringify(proposalData)}).success(function(data) {
+    $http.patch(PROPOSAL_URI + proposal.id + '/', {'data':JSON.stringify(proposalData)}).success(function(data) {
```
Comment (catherinedevlin) [inline sbirez/static/js/services/proposalsvc.js]: Is it intentional that the /partial/ was removed here?  If the existence of saveCompleteData implies that saveProposalData is supposed to save partial submissions successfully, I'm pretty sure /partial/ is what you want...

Comment (DavidEBest) [inline sbirez/static/js/services/proposalsvc.js]: Good call.

File: tests/client/spec/controllers/search.js [hunk 2]
```diff
@@ -153,20 +153,20 @@ describe('Controller: SearchCtrl', function () {
     expect(scope.numFound).toEqual(longData.count);
   });
 
-  it('saving a valid opportunity when logged in should result in a saved opportunity', function() {
+  xit('saving a valid opportunity when logged in should result in a saved opportunity', function() {
```
Comment (catherinedevlin) [inline tests/client/spec/controllers/search.js]: Disabling these tests == intentional?

Comment (DavidEBest) [inline tests/client/spec/controllers/search.js]: Yes. They need to be moved (in process now).

**Pull request merged**